web/src/utils/network.ts

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) [2022-2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import ipaddr from "ipaddr.js";
+import {
+  APIRoute,
+  ApFlags,
+  ApSecurityFlags,
+  Connection,
+  ConnectionState,
+  Device,
+  IPAddress,
+  Route,
+  SecurityProtocols,
+} from "~/types/network";
+
```

Comment:
I have a doubt with types in general that it is quite visible in the return types of functions in this file. Should we specify them always even when the typing system is able to infer it?